src/middlewared/middlewared/plugins/cloud/crud.py

```diff
@@ -99,3 +100,8 @@ async def _validate(self, verrors, name, data):
                                              ["type", "=", "FILESYSTEM"]]):
                 verrors.add(f"{name}.snapshot", "This option is only available for datasets that have no further "
                                                 "nesting")
+
+        if app and not credential_has_full_admin(app.authenticated_credentials):
+            for k in ["pre_script", "post_script"]:
+                if data[k]:
+                    verrors.add(f"{name}.{k}", "You don't have permissions to use this field")
```

Comment:
It would better to explain precise restriction:

```
The ability to edit cloud sync pre and post scripts is limited to users who have full administrative credentials.
```